### PR TITLE
docs(user-docs): add tsup migration explanation

### PR DESCRIPTION
**Pull request by fargito**

### CODE REVIEW

File: user-docs/documentation/docs/how-to-guides/3-migration-guides/2-babel-to-tsup.md [hunk 1]
```diff
@@ -6,7 +6,9 @@ sidebar_position: 2
 
 ## Why?
 
-Coming soon...
+Swarmion used [Babel](https://babeljs.io/) to transpile internal library code. Babel made it possible to transform our Typescript code to plain Javascript (both CommonJS and ESModules).
+
+However, we decided to migrate to [Tsup](https://tsup.egoist.dev/). It uses [Esbuild](https://esbuild.github.io/) and is much more efficient to transpile libraries.
```
Comment (adriencaccia) [inline user-docs/documentation/docs/how-to-guides/3-migration-guides/2-babel-to-tsup.md:11]: A link to the migration PR could be useful here 😉

**Review by adriencaccia — CHANGES_REQUESTED** [review]

File: user-docs/documentation/docs/how-to-guides/3-migration-guides/index.mdx [hunk 2]
```diff
@@ -2,10 +2,8 @@ import DocCardList from '@theme/DocCardList';
 
 # Migration guides
 
-Swarmion may seem like a fine match if you're starting a new project, but what if you already have an existing project?
+Swarmion may seem like a fine match if you're starting a new project, but what if you already have an existing project? Or what if you started from a Swarmion template, but want to benefit of new features.
```
Comment (guillaumeduboc) [inline user-docs/documentation/docs/how-to-guides/3-migration-guides/index.mdx:5]:
```suggestion
Swarmion may seem like a fine match if you're starting a new project, but what if you already have an existing project? Or what if you started from a Swarmion template but want to benefit from new features.
```

File: user-docs/documentation/docs/how-to-guides/3-migration-guides/2-babel-to-tsup.md (hunk 1 above)
Comment (fargito) [reply user-docs/documentation/docs/how-to-guides/3-migration-guides/2-babel-to-tsup.md:11]: Done

File: user-docs/documentation/docs/how-to-guides/3-migration-guides/index.mdx (hunk 2 above)
Comment (fargito) [reply user-docs/documentation/docs/how-to-guides/3-migration-guides/index.mdx]: Done

**Review by adriencaccia — APPROVED** [review]

**Pull request merged**